pyvista/core/utilities/geometric_objects.py

```diff
@@ -434,7 +434,7 @@ def Plane(
 
     surf.points[:, 0] *= i_size
     surf.points[:, 1] *= j_size
-    surf.rotate_y(-90, inplace=True)
+    surf.rotate_y(90, inplace=True)
```

Comment:
Thanks for this. I also checked the other functions in the same file and found that the `Sphere` function also had a `-90` degree rotation. Please correct the angle of the Sphere function as well if it is not affected.

Reply:
Ok. I spent some time evaluating your suggestion, and I agree with your proposed changes for `Sphere`. I also think a separate issue should be raised to clarify the meaning of `direction` in the `Sphere` docstring. Details below:

In the `Plane` case it was clear that the normal was pointing in the wrong direction because the meaning of normal or direction for a plane is clear. However, for a sphere, this is less clear. What should direction mean in this case? The docs for [vtkSphereSource](https://vtk.org/doc/nightly/html/classvtkSphereSource.html#ae25246b3c5f43545670efc7f13b10f2d) does not have an explicit `direction` parameter but it does state that "latitude angle 0 is at north pole."  Also note that the docs for [Sphere](https://docs.pyvista.org/version/0.40/api/utilities/_autosummary/pyvista.Sphere.html) define `direction` as "Direction the top of the sphere points to in [x, y, z]." So, in this case, if +Z points north and -Z points south, we can say that the sphere's direction points from its south pole to its north pole.

So, the question that should be asked is, do the current transformations  in `Sphere` correctly orient the sphere so that its direction points from south pole to north pole where the north pole is at latitude 0. To test this, we can run:
``` python
import pyvista
north = pyvista.Sphere(start_phi=0, end_phi=90)
south = pyvista.Sphere(start_phi=90, end_phi=180)
p = pyvista.Plotter()
p.add_mesh(north, show_edges=True, color='red', label='north')
p.add_mesh(south, show_edges=True, color='blue', label='south')
p.add_axes()
p.add_legend()
p.show()
```
Where the northern hemisphere is defined as theta ranging from 0-90, and the southern hemisphere is defined by theta ranging from 90-180.

With `rotate_y(-90)` we get:
![image](https://github.com/pyvista/pyvista/assets/89109579/645cfe11-e290-4848-9100-870d011277d5)
Which shows us that the southern hemisphere is above the northern hemisphere, and therefore its actual direction is (0, 0, -1). However, the direction should indeed be (0, 0, 1).

With `rotate_y(90)`, we get the correct orientation:
![image](https://github.com/pyvista/pyvista/assets/89109579/bc34f422-8ec8-4438-8e19-29172f52fc79)

The `Sphere` docs could be improved by:

- clarifying that latitude angle 0 is at north pole when setting `phi`
- clarifying that `direction` is defined from south pole to north pole ("Direction the top of the sphere points" is ambiguous in my view)


@tkoyama010 do you still want me to update this PR and change `rotate_y(-90)` to `rotate_y(90)` in `Sphere`? If so, do you also want me to clarify the documentation for `Sphere` as I suggest above? Or, would you rather keep this PR as-is, and keep it restricted to `Plane` fixes only, and then separately raise a new issue to fix the problems with `Sphere`.

Reply:
@user27182 Thank you. The relevant part was added by [this commit five years ago](https://github.com/pyvista/pyvista/commit/5254e94f5d87c3549f751e680569c07df5956e28), and there is no rational reason why the NS should not be under the Z-axis, so please add the change. Please clarify the specification by the test code and change the title of the pull request.

Reply:
I pushes these changes to `Sphere`, however this caused `test_find_cells_intersecting_line` to fail since that test made an assumption about the cell configuration based on the previous (wrong) orientation of `Sphere`.

Therefore, I pushed an additional commit to fix this test and remove its dependency on `Sphere`.

Reply:
@tkoyama010 since `Sphere` is used by so many examples and test cases, fixing the direction bug for `Sphere` seems like a massive undertaking that I am not currently unable to commit to. The changes and effort needed to fix this is likely compounded by #4691.

How would you like to proceed? Should we keep this PR for Plane only just to get that fix merged and open a separate issue specifically to address the `Sphere` direction bug?

Reply:
@user27182
Let's merge only `Plane` case and discuss Sphere in an issue. I'm sorry, but could you please revert the Sphere part and change the title back as well?

Reply:
Ok. I had a closer look at some of the tests that were failing and was able to resolve a handful of failed doctests and a few regular test cases. So, on second thought I decided to push further and continue working to resolve the problems caused by fixing `Sphere` and will also help resolve #4691 with this PR too.

However, I will need some assistance making sense of the current failed checks, especially plotting / image regression testing related failures. @tkoyama010 if you can provide some guidance on a possible cause or a path forward that would be appreciated.

Reply:
@user27182 
There seems to be an error in the regression testing of the output image, could you please refer to [Notes Regarding Image Regression Testing](https://docs.pyvista.org/version/stable/extras/developer_notes.html#notes-regarding-image-regression-testing) to check how much difference is being made to the image?

Reply:
Some of the image regression testing errors were caused by the test cases using `Plane` or `Sphere` objects. Since the cached mages were generated using the incorrect orientation of these objects, the new images generated during testing would appear flipped or mirrored when compared to what was expected, and hence the tests would fail. I have pushed new changes to these tests so that the tests now "invert" the direction of any `Plane` or `Sphere` objects (e.g. use `Sphere(direction=(0,0,-1)` instead of `Sphere()`) such that the generated images should now appear identical to the expected cached images. This resolved many failed tests locally on my machine, let's see how the CI tests fare.